x/config/mapdecode.go

```diff
@@ -35,46 +34,47 @@ const (
 	_interpolateOption = "interpolate"
 )
 
-func decodeInto(dst, src interface{}) error {
-	return mapdecode.Decode(dst, src,
-		mapdecode.TagName(_tagName),
-		mapdecode.FieldHook(interpolateHook))
+func decodeInto(dst interface{}, src interface{}, opts ...mapdecode.Option) error {
+	opts = append(opts, mapdecode.TagName(_tagName))
```

Comment:
It doesn't have to be the last one. It's just easier to append it to the end of the existing slice (from varargs) rather than creating a new slice with that option in the front and appending `opts` into it.

    opts = append([]mapdecode.Option{mapdecode.TagName(_tagName)}, opts...)